### PR TITLE
Remove async-net dependency

### PR DESCRIPTION
**Pull request by gbbosak**
Remove the dependency on async-net

### CODE REVIEW

File: mock-omaha-server/src/lib.rs [hunk 1]
```diff
@@ -29,8 +30,6 @@ use url::Url;
 
 #[cfg(feature = "tokio")]
 use {
-    async_net::TcpListener,
```
Comment (JayXon) [inline mock-omaha-server/src/lib.rs:32]: also remove the one below at line 45?

File: mock-omaha-server/src/lib.rs [hunk 2]
```diff
@@ -21,6 +21,7 @@ use omaha_client::cup_ecdsa::PublicKeyId;
 use serde::Deserialize;
 use serde_json::json;
 use sha2::{Digest, Sha256};
+use tokio::net::{TcpListener, TcpStream};
```
Comment (JayXon) [inline mock-omaha-server/src/lib.rs]: probably should put this behind `#[cfg(feature = "tokio")]`?

File: mock-omaha-server/src/lib.rs [hunk 3]
```diff
@@ -318,6 +329,8 @@ impl OmahaServer {
         arc_server: Arc<Mutex<OmahaServer>>,
         addr: Option<SocketAddr>,
     ) -> Result<(String, Option<JoinHandle<()>>), Error> {
+        use tokio::net::TcpListener;
```
Comment (JayXon) [inline mock-omaha-server/src/lib.rs]: is this still needed?

**Review by JayXon — COMMENTED** [review]
this code has a lot of different cfg, please make sure it still builds with `fasync` on fuchsia

File: mock-omaha-server/src/lib.rs (hunk 3 above)
Comment (gbbosak) [reply mock-omaha-server/src/lib.rs]: Doesn't seem to be, removed.

**Review by gbbosak — COMMENTED** [review]
Updated code to address comments. Please take another look.

**Comment by gbbosak** [comment]
Also on Fuchsia (target_os="fuchsia"), not host btw -- we aren't using async_net to begin with, it's using the fuchsia_async abstraction instead, so this patch shouldn't break target. It also shouldn't break host because we aren't changing the public interface.

**Review by JayXon — APPROVED** [review]

**Pull request merged**